modules/tinymce/src/core/test/ts/webdriver/PageUpDownKeysTest.ts

```diff
@@ -0,0 +1,115 @@
+        this.skip();
+      }
+
+      const editor = hook.editor();
+      editor.setContent('<p>text<a href="google.com">link</a>link</p>');
+      TinySelections.setCursor(editor, [ 0, 1, 0 ], 2);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.combo({}, 'PageUp') ]);
+
+      TinyAssertions.assertCursor(editor, [ 0, 1, 0 ], 2);
+    });
+
+    it('TINY-4612: caret wont move inline_boundaries: false', async function () {
+      if (platform.os.isWindows()) {
+        this.skip();
+      }
+
+      const editor = hook.editor();
+      editor.settings.inline_boundaries = false;
+      editor.setContent('<p>text<a href="google.com">link</a>link</p>');
+      TinySelections.setCursor(editor, [ 0, 1, 0 ], 0);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.combo({}, 'PageUp') ]);
+
+      TinyAssertions.assertCursor(editor, [ 0, 0 ], 4);
+      delete editor.settings.inline_boundaries;
+    });
+  });
+
+  context('Page Down', () => {
+    it('TINY-4612: caret should be placed after the inline element if it is at the start of it', async () => {
+      const editor = hook.editor();
+      editor.setContent('<p>text<a href="google.com">link</a></p>');
+      TinySelections.setCursor(editor, [ 0, 1, 0 ], 0);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.combo({}, 'PageDown') ]);
+
+      TinyAssertions.assertCursor(editor, [ 0 ], 2);
+    });
+
+    it('TINY-4612: caret should be placed after the inline element if it is at the end of it', async () => {
+      const editor = hook.editor();
+      editor.setContent('<p>text<a href="google.com">link</a></p>');
+      TinySelections.setCursor(editor, [ 0, 1, 0 ], 4);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.combo({}, 'PageDown') ]);
+
+      TinyAssertions.assertCursor(editor, [ 0 ], 2);
+    });
+
+    it('TINY-4612: caret wont move if it is not at the beginning/end of the inline element', async function () {
+      // Windows os default behavior (regardless the browser) is moving the caret at the end of the line
+      if (platform.os.isWindows()) {
+        this.skip();
+      }
+
+      const editor = hook.editor();
+      editor.setContent('<p><a href="google.com">link</a>text</p>');
+      TinySelections.setCursor(editor, [ 0, 0, 0 ], 2);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.combo({}, 'PageDown') ]);
+
+      TinyAssertions.assertCursor(editor, [ 0, 0, 0 ], 2);
+    });
+
+    it('TINY-4612: caret wont move inline_boundaries: false', async function () {
+      if (platform.os.isWindows()) {
+        this.skip();
+      }
+
+      const editor = hook.editor();
+      editor.settings.inline_boundaries = false;
+      editor.setContent('<p><a href="google.com">link</a>text</p>');
+      TinySelections.setCursor(editor, [ 0, 0, 0 ], 0);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.combo({}, 'PageDown') ]);
+
+      TinyAssertions.assertCursor(editor, [ 0, 0, 0 ], 0);
```

Comment:
@camiloHimura here is one part related to what I mentioned in the other comment. This is pressing page down, so it shouldn't be staying in the same location. It should be getting moved to `[ 0, 1 ], 4`. It's possible this may just be webdriver not working, however this assertion is incorrect.

Reply:
To clarify in case the above wasn't clear, it should be at the end of the line, after `text` (with macOS likely being the exception since as mentioned previously it scrolls the viewport).

Reply:
Updated